### ISSUE TITLE
Porter does not handle parameters of type number even though it is stated in documentation

### ISSUE THREAD

**Issue by jasmdk** [getporter/porter#1554]
## Describe the bug
https://porter.sh/author-bundles/#parameters documents the parameter type as if "number" is a valid type, but if you try to define such a parameter, porter will fail.

Also here it states that porter schema differs from CNAB as CNAB does not support non-integers (neither does Porter apparently): https://porter.sh/author-bundles/#parameter-and-output-schema

The schema (https://github.com/cnabio/cnab-spec/blob/main/schema/definitions.schema.json) also does not include 'number' as an allowed value in simpleTypes.

Also the parameter test does not cover numbers either: https://github.com/getporter/porter/blob/784d61cf283bbd5b9c05a42ea718d3d042803305/pkg/cnab/provider/parameters_test.go#L61

## To Reproduce
1. Run `porter create`
2. Add a parameter of type number: 
```
parameters:
  - name: some_number
    type: number
```
3. Run this porter command `porter install --param some_number=1.0`

## Expected behavior
Porter should not fail, but print the "Hello World" message

## Porter Command and Output
```
$ porter install --param some_number=1
installing porter-hello...
Error: invalid parameters: unable to convert parameter's some_number value 1 to the destination parameter type number: invalid definition
```

## Version
porter v0.37.2 (26299559)

**Comment by carolynvs** [comment]
Thanks for clearly identifying where the documentation is out of date. I see two problems:

1. Our documentation is incorrect. What works right now with porter is the "integer" parameter type.
2. After reviewing the canonical json spec, the cnab spec and the recommended library for canonical json in Go, I believe we made a mistake in the cnab spec limiting numbers. I have opened an issue with the cnab spec to discuss it further. https://github.com/cnabio/cnab-spec/issues/413
